main.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"bytes"
+	"encoding/xml"
+	"flag"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+const (
+	namespace = "passenger_nginx"
+)
+
+var (
+	Version = "0.0.0"
+)
+
+// Exporter collects metrics from a passenger-nginx integration.
+type Exporter struct {
+	// binary file path for querying passenger state.
+	binPath string
+	args    []string
+
+	// Passenger metrics.
+	up                  *prometheus.Desc
+	uptime              *prometheus.Desc
+	version             *prometheus.Desc
+	toplevelQueue       *prometheus.Desc
+	maxProcessCount     *prometheus.Desc
+	currentProcessCount *prometheus.Desc
+	appCount            *prometheus.Desc
+
+	// App metrics.
+	appQueue         *prometheus.Desc
+	appProcsSpawning *prometheus.Desc
+
+	// Process metrics.
+	requestsProcessed *prometheus.Desc
+	procUptime        *prometheus.Desc
+	procMemory        *prometheus.Desc
+	procStatus        *prometheus.Desc
+	// There are several Process level metrics that could be of interest,
+	// but I imagine they're already handled by node_exporter? Not sure
+	// on their granularity. e.g. passenger status exposes swap memory
+	// usage on the process, need to check if node_exporter allows
+	// per-process swap metrics.
+}
+
+func NewExporter(cmd string) (*Exporter, error) {
+	cmdComponents := strings.Split(cmd, " ")
+	binPath := cmdComponents[0]
+
+	_, err := os.Stat(binPath)
```

Comment:
K, yeah I was wondering that, but thought that maybe we would expect people to run the exporter with appropriate privileges.

Will remove the stat check.